### ISSUE TITLE
Excessively slow BN1d on gpu for large batch size.

### ISSUE THREAD

**Issue by mfinzi** [pytorch/pytorch#61587]
## 🐛 Bug

nn.BatchNorm1d is unusually slow on GPU for inputs around the size (100000,32)
whereas the naïve implementation (adapted from https://github.com/ptrblck/pytorch_misc/blob/master/batch_norm_manual.py) runs about 20x faster.

```
BN1d Forward: 6.292s
Naive BN1d Forward: 0.320s
```

Likely related to the closed issue [#12006](https://github.com/pytorch/pytorch/issues/12006), but perhaps there has been some regression or the previous problem was never fully resolved.

## Code Example To Reproduce

```python
import torch
device = "cuda"

# Time pytorch BN1d
start = torch.cuda.Event(enable_timing=True)
end = torch.cuda.Event(enable_timing=True)

x = torch.randn(100000,32).to(device)
bn = torch.nn.BatchNorm1d(32).to(device)

N=100
start.record()
for i in range(N):
    out = bn(x)
end.record()

torch.cuda.synchronize()
print("BN1d Forward:", start.elapsed_time(end)/N)
```
```python
# Time Naive implementation
start = torch.cuda.Event(enable_timing=True)
end = torch.cuda.Event(enable_timing=True)

x = torch.randn(100000,32).to(device)
bn = MyBN1(32).to(device)

N=100
start.record()
for i in range(N):
    out = bn(x)
end.record()

torch.cuda.synchronize()
print("Naive BN1d Forward:",start.elapsed_time(end)/N)
```

## Environment

PyTorch version: 1.8.1+cu102
Is debug build: False
CUDA used to build PyTorch: 10.2
ROCM used to build PyTorch: N/A

OS: Ubuntu 18.04.2 LTS (x86_64)
GCC version: (Ubuntu 7.4.0-1ubuntu1~18.04.1) 7.4.0
Clang version: Could not collect
CMake version: version 3.16.4
Libc version: glibc-2.25

Python version: 3.6.9 (default, Apr 18 2020, 01:56:04)  [GCC 8.4.0] (64-bit runt
ime)
Python platform: Linux-4.15.0-55-generic-x86_64-with-Ubuntu-18.04-bionic
Is CUDA available: True
CUDA runtime version: 10.2.152
GPU models and configuration: GPU 0: NVIDIA TITAN V
Nvidia driver version: 465.27
cuDNN version: /usr/lib/x86_64-linux-gnu/libcudnn.so.7.6.6
HIP runtime version: N/A
MIOpen runtime version: N/A

cc @albanD @mruberry @jbschlosser @VitalyFedyunin @ngimel @heitorschueroff

**Comment by ngimel** [comment]
Please update to verion 1.9/nightly, on V100 I'm getting 
```
BN1d Forward: 0.20148639678955077

```

**Comment by mfinzi** [comment]
It does seem to be fixed in the new version, thanks @ngimel.
